src/jtagmkI.c

```diff
@@ -1075,6 +1075,31 @@ static int jtagmkI_set_sck_period(const PROGRAMMER *pgm, double v) {
 }
 
 
+static int jtagmkI_get_sck_period(const PROGRAMMER *pgm, double *v) {
+  unsigned char dur = 0;
+  if (jtagmkI_getparm(pgm, PARM_CLOCK, &dur) < 0)
+    return -1;
+  if (dur == JTAG_BITRATE_1_MHz)
+    *v = 1e6;
+  else if (dur == JTAG_BITRATE_500_kHz)
+    *v = 500e3;
+  else if (dur == JTAG_BITRATE_250_kHz)
+    *v = 250e3;
+  else
```

Comment:
Would it not be better to have sth like
```
  else if(dur == JTAG_BITRATE_125_kHz)
    *v = 125e3;
  else
     return -1;
```

AVRDUDE is normally good at checking all possible things that can go wrong. We might also want to sprinkle in a `pmsg_error("(reason for returning -1)\n");` when a function couldn't do its job and returns -1. This makes debugging soooo much easier.